drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/cde_export.rb

```diff
@@ -0,0 +1,64 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module HmisExternalApis::AcHmis::Exporters
+  class CdeExport
+    attr_accessor :output
+
+    def initialize(output = StringIO.new)
+      require 'csv'
+      self.output = output
+    end
+
+    def run!
+      Rails.logger.info 'Generating CDE report'
+      write_row(columns)
+      total = cdes.count
+      Rails.logger.info "There are #{total} CDEs to export"
+
+      cdes.each.with_index do |cde, i|
+        Rails.logger.info "Processed #{i} of #{total}" if (i % 100).zero?
+        values = [
+          cde.id,
+          cde.data_element_definition.key,
+          cde.data_element_definition.owner_type.demodulize,
+          cde.owner.id,
```

Comment:
We should preload `owner` on `cdes` so that this doesn't do a db lookup for each iteration

Reply:
Oops, really the faster way to do this would be `cde.owner_id`. That way it does not have to hit the database to load the owner records at all. That is probably worth a change.

Reply:
Great, thank you :D